shared/styles/mixins.scss

```diff
@@ -77,19 +76,42 @@
   }
 }
 
+@mixin v-scale($property, $number) {
+  #{$property}: $number;
+
+  @supports (--css: variables) {
+    #{$property}: calc(#{$number} * var(--scale-element));
+  }
+}
+
 // segment customisable per-component / instance rather than being forced to work around defaults.
 @mixin segment($padding-top: $segment-padding, $padding-bottom: $segment-padding, $padding-top-mobile: $segment-padding-mobile, $padding-bottom-mobile: $segment-padding-mobile) {
-  padding-top: $padding-top-mobile;
-  padding-bottom: $padding-bottom-mobile;
+  padding-top: strip-unit($padding-top-mobile) + px;
+  padding-bottom: strip-unit($padding-bottom-mobile) + px;
+
+  @supports (--css: variables) {
+    padding-top: calc(#{strip-unit($padding-top-mobile) + px} * var(--scale-element));
```

Comment:
Maybe use the v-scale mixin here? It could also handle the unit thing, making things look cleaner.
Although the output would have more `@supports (--css: variables)`lines I guess...

Any way for us to do something like:
```scss
@include v-scale {
  padding-top: $padding-top-mobile;
  padding-bottom: $padding-bottom-mobile;
}
```
Where it somehow loops over `@content` and adds all the calculations?

Reply:
yeah thats a good idea also. will add in.